### PR TITLE
Updated Fix for Deployment defaulting to first item when filter filters out all items

### LINKED ISSUE #12851: Depoyment defaults to first item when filter filters out all items
When filtering items in the resource deployment dialog if you type in a query that filters out all resources then it'll still have the "first" item selected silently in the background and allow the user to continue. The continue button should be disabled if there aren't any visible selected resources.

Capture

### PR DESCRIPTION
**Pull request by aasimkhan30**
This PR fixes #12851

### CODE REVIEW

File: extensions/resource-deployment/src/ui/resourceTypePickerDialog.ts [hunk 1]
```diff
@@ -266,7 +266,9 @@ export class ResourceTypePickerDialog extends DialogBase {
 			this._agreementCheckboxChecked = false;
 			this._agreementContainer.clearItems();
 			this._optionsContainer.clearItems();
-			this.updateToolsDisplayTable();
+			this._toolsLoadingComponent.loading = false;
+			this._toolsTable.data = [[localize('deploymentDialog.NoRequiredTool', "No tools required"), '']];
```
Comment (Charles-Gagnon) [inline extensions/resource-deployment/src/ui/resourceTypePickerDialog.ts:270]: Please include a screenshot - I'm not sure how this sentence makes sense when there's no options to even select

Comment (aasimkhan30) [reply extensions/resource-deployment/src/ui/resourceTypePickerDialog.ts:270]: image

Comment (Charles-Gagnon) [reply extensions/resource-deployment/src/ui/resourceTypePickerDialog.ts:270]: I don't like having an empty table there to begin with - but removing it is going to annoying I think. 

If you don't have a row what does it look like? That'd be slightly preferable I think.

Comment (aasimkhan30) [reply extensions/resource-deployment/src/ui/resourceTypePickerDialog.ts:270]: image

Yeah, this looks better

**Review by Charles-Gagnon — APPROVED** [review]

**Pull request merged**